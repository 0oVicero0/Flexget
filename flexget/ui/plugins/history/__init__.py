from __future__ import unicode_literals, division, absolute_import

from flexget.ui import register_plugin, register_js, Blueprint, register_menu

history = Blueprint('history', __name__)
register_plugin(history)

history.register_angular_route(
    '',
    url=history.url_prefix,
    template_url='history.html',
    controller='HistoryCtrl',
)

register_js('history', 'js/history.js', bp=history)

register_menu(history.url_prefix, 'History', icon='fa fa-history')
