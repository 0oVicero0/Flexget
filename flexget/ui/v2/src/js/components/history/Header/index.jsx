import React, { Component } from 'react';
import SecondaryNav from 'components/common/SecondaryNav';

class Header extends Component {
  componentDidMount() {

  }

  render() {
    return (
      <SecondaryNav>
        <div />
      </SecondaryNav>
    );
  }
}

export default Header;
