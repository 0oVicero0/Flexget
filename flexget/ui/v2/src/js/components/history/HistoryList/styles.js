import styled from 'emotion/react';
import { ListSubheader } from 'material-ui/List';

export const Subheader = styled(ListSubheader)`
  background-color: white;
`;

export const Wrapper = styled.div`
  overflow-y: scroll;
  height: calc(100% + 2rem);
  margin: -1rem;
`;
