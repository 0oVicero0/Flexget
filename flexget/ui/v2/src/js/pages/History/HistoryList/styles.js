import styled from 'emotion/react';
import { ListSubheader } from 'material-ui/List';

export const Subheader = styled(ListSubheader)`
  background-color: white;
`;

export const Wrapper = styled.div`
  overflow-y: scroll;
  flex: 1;
`;
