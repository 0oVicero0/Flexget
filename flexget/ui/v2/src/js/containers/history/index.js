import { connect } from 'react-redux';
import HistoryPage from 'components/history';
import { request } from 'utils/actions';
import { GET_HISTORY } from 'store/history/actions';

function mapDispatchToProps(dispatch) {
  return {
    getHistory: data => dispatch(request(GET_HISTORY, data)),
  };
}

export default connect(null, mapDispatchToProps)(HistoryPage);
