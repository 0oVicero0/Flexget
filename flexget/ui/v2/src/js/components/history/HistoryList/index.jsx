import React, { Component } from 'react';
import PropTypes from 'prop-types';
import InfiniteScroll from 'react-infinite-scroller';
import List, { ListItem, ListItemText } from 'material-ui/List';
import LoadingSpinner from 'components/common/LoadingSpinner';
import { Subheader, Wrapper } from './styles';

class HistoryList extends Component {
  static propTypes = {
    grouping: PropTypes.oneOf(['time', 'task']).isRequired,
    history: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.shape({
      task: PropTypes.string,
      title: PropTypes.string,
      url: PropTypes.string,
      filename: PropTypes.string,
      details: PropTypes.string,
      time: PropTypes.strings,
      id: PropTypes.number,
    }))).isRequired,
    hasMore: PropTypes.bool,
    getHistory: PropTypes.func.isRequired,
  };

  static defaultProps = {
    hasMore: true,
  };

  render() {
    const { grouping, history, hasMore, getHistory } = this.props;

    return (
      <Wrapper>
        <InfiniteScroll
          hasMore={hasMore}
          loadMore={getHistory}
          loader={<LoadingSpinner loading />}
          ref={(node) => { this.scroller = node; }}
          useWindow={false}
        >
          { Object.entries(history).map(([subheader, histories]) => (
            <List key={subheader} subheader={<Subheader color="primary">{subheader}</Subheader>}>
              {histories.map(({ id, title, time, task }) => (
                <ListItem key={id}>
                  <ListItemText primary={title} secondary={grouping === 'time' ? task : time} />
                </ListItem>
              ))}
            </List>
          ))}
        </InfiniteScroll>
      </Wrapper>
    );
  }
}

export default HistoryList;
