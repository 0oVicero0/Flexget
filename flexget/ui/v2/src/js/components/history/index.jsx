import React, { Component } from 'react';
import PropTypes from 'prop-types';
import HistoryList from 'containers/history/HistoryList';
import Header from 'components/history/Header';
import { NoPaddingWrapper } from 'components/common/styles';

export default class HistoryPage extends Component {
  static propTypes = {
    getHistory: PropTypes.func.isRequired,
  };

  state = {
    grouping: 'time',
    query: {
      sort_by: 'time',
      order: 'desc',
    },
  };

  getHistory = page => this.props.getHistory({ page, ...this.state.query })

  changeGrouping = grouping => this.setState({ grouping })

  changeQuery = query => this.setState({ query })

  restartLoader = () => { this.list.scroller.pageLoaded = 0; }

  render() {
    const { grouping } = this.state;

    return (
      <NoPaddingWrapper>
        <Header
          changeQuery={this.changeQuery}
          changeGrouping={this.changeGrouping}
          restartLoader={this.restartLoader}
          {...this.state}
        />
        <HistoryList
          grouping={grouping}
          getHistory={this.getHistory}
          ref={(node) => { this.list = node; }}
        />
      </NoPaddingWrapper>
    );
  }
}
