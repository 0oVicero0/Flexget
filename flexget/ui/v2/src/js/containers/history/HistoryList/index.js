import { connect } from 'react-redux';
import HistoryList from 'components/history/HistoryList';
import { getGroupedHistory } from 'store/history/selectors';

export function mapStateToProps({ history }, { grouping }) {
  return {
    history: getGroupedHistory(history, grouping),
    hasMore: !history.totalCount || history.items.length < history.totalCount,
  };
}

export default connect(mapStateToProps)(HistoryList);
