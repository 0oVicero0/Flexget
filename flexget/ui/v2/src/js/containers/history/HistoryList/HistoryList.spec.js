import { mapStateToProps } from 'containers/history/HistoryList';

describe('containers/history/HistoryList', () => {
  it('should return the right params if there are more', () => {
    const result = mapStateToProps({
      history: {
        items: [{ task: 'task_name' }],
        totalCount: 3,
      },
    }, {
      grouping: 'task',
    });

    expect(result.history).toMatchSnapshot();
    expect(result.hasMore).toBe(true);
  });

  it('should return the right params if there are not more', () => {
    const result = mapStateToProps({
      history: {
        items: [{ task: 'task_name' }],
        totalCount: 1,
      },
    }, {
      grouping: 'task',
    });

    expect(result.history).toMatchSnapshot();
    expect(result.hasMore).toBe(false);
  });
});
