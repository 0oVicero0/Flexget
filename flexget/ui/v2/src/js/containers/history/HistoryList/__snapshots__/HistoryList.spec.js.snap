// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/history/HistoryList should return the right params if there are more 1`] = `
Object {
  "task_name": Array [
    Object {
      "task": "task_name",
    },
  ],
}
`;

exports[`containers/history/HistoryList should return the right params if there are not more 1`] = `
Object {
  "task_name": Array [
    Object {
      "task": "task_name",
    },
  ],
}
`;
